django_irods/storage.py

```diff
@@ -308,9 +308,58 @@ def listdir(self, path):
                     listing[2].append(size)
         return listing
 
+    def listdir(self, path):
+        # the query below returns name and size (separated in comma) of all data
+        # objects/files under the path collection/directory
+        qrystr = "select DATA_NAME, DATA_SIZE where DATA_REPL_STATUS != '0' AND " \
+                 "COLL_NAME like '%{}'".format(path)
+        stdout = self.session.run("iquest", None, "--no-page", "%s,%s",
+                                  qrystr)[0].split("\n")
+
+        listing = ([], [], [])
+
+        for i in range(len(stdout)):
+            if not stdout[i] or "CAT_NO_ROWS_FOUND" in stdout[i]:
+                break
+            file_info = stdout[i].rsplit(',', 1)
+            listing[1].append(file_info[0])
+            listing[2].append(file_info[1])
+
+        # the query below returns name of all sub-collections/sub-directories
+        # under the path collection/directory
+        qrystr = "select COLL_NAME where COLL_PARENT_NAME like '%{}'".format(path)
```

Comment:
There are two queries... for readability, I think it makes sense to put them into different methods (list_files, list_directories) and then call and merge the results in this method.

Reply:
Done. See if it looks good now.